pontoon/base/models/entity.py

```diff
@@ -855,10 +856,19 @@ def for_project_locale(
             # Modify query based on case sensitivity filter
             translation_filters = (
                 (
-                    Q(translation__string__contains=s)
-                    if search_match_case
-                    else Q(
-                        translation__string__icontains_collate=(s, locale.db_collation)
+                    Q(translation__string__regex=rf"\y{s}\y")
+                    if search_match_word and search_match_case
+                    else Q(translation__string__iregex=rf"\y{s}\y")
+                    if search_match_word
+                    else (
+                        Q(translation__string__contains=s)
+                        if search_match_case
+                        else Q(
+                            translation__string__icontains_collate=(
+                                s,
+                                locale.db_collation,
+                            )
+                        )
```

Comment:
If you declared this before line 855:
```python
i = "" if search_match_case else "i"
y = "" if search_match_word else r"\y"
```
You could replace this whole block with this:
```python
Q(**{f"translation__string__{i}regex": rf"{y}{s}{y}"})
```

We can drop `icontains_collate`, because it doesn't work anyways apparently (#3323). And to fix that bug, we might want to implement regex_collate instead.